design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +235,134 @@ permissionScope: Cluster
 
 ### Stack Install
 
-Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to allow differing access to request their installation.
-A simple way to approach this would be to create different installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to
+ allow differing access to request their installation. A simple way to approach this would be to create different
+ installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+
 Currently, we use a single `StackRequest` type, but going forward we should differentiate by stack type:
 
-* `ClusterStackInstall`: requests the installation of a cluster stack
-* `StackInstall`: requests the installation of a namespace stack
+* `ClusterStackInstall`: requests the installation of a cluster stack who's resources are all cluster scoped
+* `StackInstall`: requests the installation of a namespace stack who's resources are all namespace scoped
+
+With these two separate types, an administrator can allow access to installing cluster stacks to only certain restricted
+ roles, then perhaps be more permissive with access to installing namespaced stacks.
+
+If the install scope type does not match what the stack author has declared within their stack metadata, an error status
```

Comment:
What in the stack metadata will identify it as a cluster-scoped or namespace-scoped stack? Inferred from stack metadata that says owned stack CRDs should be added to a `Role` or `ClusterRole` for each of the supported roles (admin, editor, viewer)?

Reply:
Within the `app.yaml` there is a permissionScope field. This is either `Cluster` or `Namespaced` and the StackManager verifies that the appropriate installation type was used (StackInstall, ClusterStackInstall).